K2Bridge/RewriteRules/RewriteSearchRule.cs

```diff
@@ -18,52 +18,21 @@ internal class RewriteSearchRule : IRule
-        /// to the IndexListController.
+        /// Requests to /_resolve/index are sent to IndexListController.
         /// We ignore /.kibana/* routs as it's internal for Kibana.
         /// </summary>
         /// <param name="context">The context object which holds the request path.</param>
-        public async void ApplyRule(RewriteContext context)
+        public void ApplyRule(RewriteContext context)
         {
             if (context.HttpContext.Request.Path.Value.Contains("_search", System.StringComparison.OrdinalIgnoreCase)
                 && !context.HttpContext.Request.Path.Value.Contains(".kibana", System.StringComparison.OrdinalIgnoreCase))
             {
-                // enable buffering to read the body stream multiple times.
-                context.HttpContext.Request.EnableBuffering();
-
-                using var reader = new StreamReader(
-                    context.HttpContext.Request.Body,
-                    encoding: Encoding.UTF8,
-                    detectEncodingFromByteOrderMarks: false,
-                    bufferSize: 8 * 1024,
-                    leaveOpen: true);
-
-                var body = await reader.ReadToEndAsync();
-
-                // Reset the request body stream position so the next middleware can read it
-                context.HttpContext.Request.Body.Position = 0;
-
-                // Although it doesn't make sense for a search query to have an empty body, we let it pass here and fail it later own when the body is analyzed.
-                if (string.IsNullOrEmpty(body))
-                {
-                    // This is a regular search (documents) request
-                    context.HttpContext.Request.Path = $"/Query/SingleSearch/{GetIndexNameFromPath(context.HttpContext.Request.Path)}";
-                    return;
-                }
-
-                JObject jo = JObject.Parse(body);
-                var aggsIndices = jo.SelectToken("aggs.indices.terms.field");
-
-                if (aggsIndices != null)
-                {
-                    // This is a request for the index list
-                    context.HttpContext.Request.Path = $"/IndexList/Process/{GetIndexNameFromPath(context.HttpContext.Request.Path)}";
-                }
-                else
-                {
-                    // This is a regular search (documents) request
-                    context.HttpContext.Request.Path = $"/Query/SingleSearch/{GetIndexNameFromPath(context.HttpContext.Request.Path)}";
-                }
+                context.HttpContext.Request.Path = $"/Query/SingleSearch/{GetIndexNameFromPath(context.HttpContext.Request.Path)}";
+            }
+            else if (context.HttpContext.Request.Path.Value.Contains("_resolve/index", System.StringComparison.OrdinalIgnoreCase))
+            {
+                // Resolve index query from Kibana 7
+                context.HttpContext.Request.Path = $"/IndexList/Resolve/{GetIndexNameFromResolvePath(context.HttpContext.Request.Path)}";
```

Comment:
Since the route pattern has changed and became simpler in the sense it's not a subset of the parent _search route, it looks like we can remove rewrites and have the route definitions directly on the controller. Can you please have another look at this?

Reply:
👍🏻